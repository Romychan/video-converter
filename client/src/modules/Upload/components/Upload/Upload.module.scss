.container {
  max-width: 765px;
  width: 100%;
  min-height: 100vh;
  margin: auto;
  padding: 0px 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 2rem;
}
