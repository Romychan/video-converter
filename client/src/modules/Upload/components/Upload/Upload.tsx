import { useNavigate } from 'react-router-dom';

import { CONVERT_PATH } from '@router/routes';
import { ACCEPTED_VIDEO_TYPES } from '../../constants/acceptedTypes';
import { getFileExtension } from '../../utils/getFileExtension';

import { useActions } from '@hooks/useActions';

import { UploadFile } from '@components/UploadFile/UploadFile';
import { UploadTitle } from '../UploadTitle/UploadTitle';

import styles from './Upload.module.scss';

export const Upload = () => {
  const navigate = useNavigate();
  const { setInputVideo, resetSettingsStore } = useActions();

  const handleUploadFile = (file: FileList) => {
    navigate(CONVERT_PATH);

    resetSettingsStore();

    setInputVideo({
      uri: URL.createObjectURL(file[0]),
      extension: getFileExtension(file[0].name),
    });
  };

  return (
    <div className={styles.container}>
      <UploadTitle />

      <UploadFile uploadFile={handleUploadFile} types={ACCEPTED_VIDEO_TYPES} />
    </div>
  );
};
