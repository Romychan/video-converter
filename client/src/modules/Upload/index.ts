export { Upload } from './components/Upload/Upload';
