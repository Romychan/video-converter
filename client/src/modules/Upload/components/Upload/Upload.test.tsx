import {
  RenderResult,
  fireEvent,
  screen,
  waitFor,
} from '@testing-library/react';
import { vi } from 'vitest';

import { renderTestApp } from '@tests/helpers/renderTestApp';
import { Upload } from './Upload';

const mockedUsedNavigate = vi.fn();

vi.mock('react-router-dom', async () => {
  const actual = (await vi.importActual('react-router-dom')) as any;

  return {
    ...actual,
    useNavigate: () => mockedUsedNavigate,
  };
});

describe('Upload', () => {
  let container: RenderResult;

  beforeEach(() => {
    mockedUsedNavigate.mockReset();

    container = renderTestApp(<Upload />);
  });

  it('should upload file', async () => {
    const file = new File(['Filename'], 'Filename.mp4', { type: 'video/mp4' });
    const dropzoneInput = screen.getByTestId(
      'dropzone-input',
    ) as HTMLInputElement;

    await waitFor(() =>
      fireEvent.change(dropzoneInput, {
        target: { files: [file] },
      }),
    );

    expect(dropzoneInput.files?.[0]).toStrictEqual(file);
    expect(dropzoneInput.files).toHaveLength(1);

    expect(mockedUsedNavigate).toHaveBeenCalledWith('/convert');
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
