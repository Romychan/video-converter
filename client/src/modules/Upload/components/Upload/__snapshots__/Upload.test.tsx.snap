// Vitest Snapshot v1

exports[`Upload > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_fd78da"
      >
        <div
          class="_header_6c7c0d"
        >
          <h1
            class="_title_6c7c0d subtitle"
          >
            Конвертер видео
          </h1>
          <p
            class="_text_6c7c0d text"
          >
            Video Converter - это онлайн-конвертер видео, который позволяет конвертировать видеофайлы в таких форматах, как MP4, AVI, GIF, MOV и другие.
          </p>
        </div>
        <div
          class="_container_c96111"
        >
          <div
            class="_dropzone_c96111"
            data-testid="dropzone-container"
          >
            <input
              accept=".mp4, .flv, .m4v, .mov"
              class="_input_c96111"
              data-testid="dropzone-input"
              type="file"
            />
            <div
              class="_icon_c96111"
            >
              <svg
                class="icon icon-upload-video"
                data-testid="icon"
                height="16"
                stroke=""
                width="16"
              >
                <use
                  xlink:href="/src/assets/img/general/icons.svg#upload-video"
                />
              </svg>
            </div>
            <p
              class="subtitle _subtitle_c96111"
            >
              Перетащите видеофайл в эту область
            </p>
            <p
              class="text text_small _text_c96111"
            >
              или нажмите на кнопку
            </p>
            <button
              class="_button_b22f72 _button_c96111 _primary_b22f72"
            >
              Выбрать файл
            </button>
          </div>
          <div
            class="_background_c96111"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_fd78da"
    >
      <div
        class="_header_6c7c0d"
      >
        <h1
          class="_title_6c7c0d subtitle"
        >
          Конвертер видео
        </h1>
        <p
          class="_text_6c7c0d text"
        >
          Video Converter - это онлайн-конвертер видео, который позволяет конвертировать видеофайлы в таких форматах, как MP4, AVI, GIF, MOV и другие.
        </p>
      </div>
      <div
        class="_container_c96111"
      >
        <div
          class="_dropzone_c96111"
          data-testid="dropzone-container"
        >
          <input
            accept=".mp4, .flv, .m4v, .mov"
            class="_input_c96111"
            data-testid="dropzone-input"
            type="file"
          />
          <div
            class="_icon_c96111"
          >
            <svg
              class="icon icon-upload-video"
              data-testid="icon"
              height="16"
              stroke=""
              width="16"
            >
              <use
                xlink:href="/src/assets/img/general/icons.svg#upload-video"
              />
            </svg>
          </div>
          <p
            class="subtitle _subtitle_c96111"
          >
            Перетащите видеофайл в эту область
          </p>
          <p
            class="text text_small _text_c96111"
          >
            или нажмите на кнопку
          </p>
          <button
            class="_button_b22f72 _button_c96111 _primary_b22f72"
          >
            Выбрать файл
          </button>
        </div>
        <div
          class="_background_c96111"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
